rust/operator-binary/src/airflow_controller.rs

```diff
@@ -151,6 +214,27 @@ pub async fn reconcile_airflow(airflow: Arc<AirflowCluster>, ctx: Arc<Ctx>) -> R
     )
     .context(InvalidProductConfigSnafu)?;
 
+    let authentication_class = match &airflow.spec.authentication_config {
+        Some(authentication_config) => {
+            match &authentication_config.authentication_class {
+                Some(authentication_class) => {
+                    Some(
+                        client
+                            .get::<AuthenticationClass>(authentication_class, None) // AuthenticationClass has ClusterScope
```

Comment:
I've added a helper function in case you're interested https://github.com/stackabletech/operator-rs/commit/8c573874474d7f4b21719955844eb460d1f82d42
We only need a release of operator-rs for that